GDevelop.js/scripts/generate-dts.mjs

```diff
@@ -237,14 +353,91 @@ class EmscriptenObject {
+${Object.entries(castFunctions)
+  .map(
+    ([interfaceName, { returnType, inputType = 'EmscriptenObject' }]) => `
+export function as${interfaceName}(object: ${inputType}): ${returnType};
+`
+  )
+  .join('')}
+
+export const Object: typeof gdObject;
+
+/**
+ * Initialises the Platforms included in the build (currently, only the JsPlatform),
+ * and loads all built-in extensions into the platform.
+ * To be called once when the library is first loaded.
+ */
+export const initializePlatforms: typeof ProjectHelper.initializePlatforms;
+
+/**
+ * Returns the pointer in WASM memory to an object. It is a number that uniquely 
+ * represents that instance of the object.
+ * 
+ * @see {@link wrapPointer} to convert a pointer back to an object.
+ */
+export function getPointer(object: EmscriptenObject): number;
+
+type ClassConstructor<T> = {
+  new (...args: any[]): T;
+};
+
+/**
+ * Wraps a pointer with a wrapper class, allowing to use the object located at the
+ * pointer's destination as an instance of that class.
+ * 
+ * @see {@link getPointer} to get a pointer from an object.
+ */
+export function wrapPointer<T extends EmscriptenObject>(ptr: number, objectClass: ClassConstructor<T>): T;
+
+/**
+ * Casts an object to another class type.
+ * 
+ * **Careful** - this is not a conversion function. 
+ * This only changes the class type and functions exposed, not the underlying memory.
+ * Only cast to another class if you are certain that the underlying memory is of that type!
+ */
+export function castObject<T extends EmscriptenObject>(object: EmscriptenObject, objectClass: ClassConstructor<T>): T;
+
+/**
+ * Checks whether two objects are pointing to the same underlying memory.
+ * A reference to the object itself is not trustworthy, since there may be multiple
+ * wrapper objects (which allow to call C++ function on C++ memory) for a single
+ * pointer ("real object").
+ * 
+ * This function must be therefore used to check for referential equality instead of
+ * JavaScript's standard equality operators when handling Emscripten objects.
+ */
+export function compare<T extends EmscriptenObject>(object1: T, object2: T): boolean;
+
+/** 
+ * Call this to free the object's underlying memory. It may not be used afterwards. 
+ * 
+ * **Call with care** - if the object owns some other objects, those will also be destroyed, 
+ * or if this object is owned by another object that does not expect it to be externally deleted 
+ * (e.g. it is a child of a map), objects will be put in an invalid state that will most likely
+ * crash the app.
+ * 
+ * If the object is owned by your code, you should still call this method when adequate, as 
+ * otherwise the memory will never be freed, causing a memory leak, which is to be avoided.
+ * 
+ * The alias {@link EmscriptenObject.delete} is recommended instead, for readability.
+ */
+export function destroy(object: EmscriptenObject): void;
+
 export as namespace gd;
+
+declare global {
+  const gd: typeof gd;
+}
```

Comment:
Actually, this might be something we should put in a .d.ts file in newIDE if/when we switch to TS there, but not include in this file?